### PR TITLE
Fix the broken DM notification

### PR DESCRIPTION
**Pull request by SebinSong**
resolve **solution 1.** of #2386 

I can see the notification works properly now.

notification works

### CODE REVIEW

File: frontend/controller/actions/chatroom.js [hunk 1]
```diff
@@ -42,7 +44,7 @@ sbp('okTurtles.events/on', MESSAGE_RECEIVE_RAW, ({
     messageHash: msgData.hash,
     height: msgData.height,
     text: msgData.text,
-    isDMOrMention: isMentionedMe || getters.chatRoomAttributes.type === CHATROOM_TYPES.DIRECT_MESSAGE,
+    isDMOrMention: isMentionedMe || targetChatroomState.attributes.type === CHATROOM_TYPES.DIRECT_MESSAGE,
```
Comment (SebinSong) [inline frontend/controller/actions/chatroom.js]: This was the cause of the issue.
`getters.chatRoomAttributes` getter here is for a currently active chatroom which means the DM notification only works when the receivers of the DM have the chatroom open in their browsers and this is what I observed too.

Made a fix here where it checks if the target chatroom is DM or not, regardless of whether the chatroom is active or not.

**Comment by SebinSong** [comment]
@taoeffect I realized that there is a requirement to update the DM test-suite for this bug too. Can I maybe work on it as a separate PR later?

**Comment by SebinSong** [comment]
@taoeffect The PR is ready for review.

I can work on updating(or creating) a test item to check for this bug separately from this PR later.

**Review by taoeffect — APPROVED** [review]
Seems to have done it!

**Pull request merged**